src/main/java/com/consumablecooldowns/ConsumableCooldownsOverlay.java

```diff
@@ -108,63 +108,191 @@ private void renderConsumableCooldowns(Graphics2D graphics, WidgetItem widgetIte
+		Arc2D.Float arc = new Arc2D.Float(Arc2D.PIE);
+		arc.setAngleStart(90); // start at top
+		arc.setAngleExtent((1 - percent) * 360); // percentage of pie to draw
+		arc.setFrame(outerFrame);
 
-		graphics.setColor(Color.BLACK);
-		graphics.drawString(delayText, textX + 1, textY + 1);
-		graphics.setColor(config.getTextColor());
-		graphics.drawString(delayText, textX, textY);
+		graphics.setClip(arc);
+		graphics.drawImage(inventoryItemImage, (int) slotBounds.getX(), (int) slotBounds.getY(), null);
+		graphics.setClip(slotBounds); // reset clip
 	}
 
-	private void renderCooldownFill(Graphics2D graphics, WidgetItem widgetItem, Rectangle slotBounds)
+	private InventoryIconInfo getInventoryIconInfo(int itemId, int quantity, Image image)
 	{
-		if (!config.showItemCooldownFill())
+		long key = (((long) itemId) << 32) | quantity;
+		InventoryIconInfo inventoryIconInfo = inventoryIconInfoCache.getIfPresent(key);
+		if (inventoryIconInfo != null)
 		{
-			return;
+			return inventoryIconInfo;
 		}
 
-		final Image image = getFillImage(config.getItemFillColor(), widgetItem.getId(), widgetItem.getQuantity());
-		graphics.drawImage(image, (int) slotBounds.getX(), (int) slotBounds.getY(), null);
+		inventoryIconInfo = getInventoryIconInfoFromImage((BufferedImage) image);
+		inventoryIconInfoCache.put(key, inventoryIconInfo);
+		return inventoryIconInfo;
 	}
 
-	private Image getFillImage(Color color, int itemId, int qty)
+	private Image getFillImage(Color color, int opacity, int itemId, int qty)
 	{
 		long key = (((long) itemId) << 32) | qty;
 		Image image = fillCache.getIfPresent(key);
 		if (image == null)
 		{
-			final Color fillColor = ColorUtil.colorWithAlpha(color, config.getItemFillOpacity());
+			final Color fillColor = ColorUtil.colorWithAlpha(color, opacity);
 			image = ImageUtil.fillImage(itemManager.getImage(itemId, qty, false), fillColor);
 			fillCache.put(key, image);
 		}
 		return image;
 	}
+
+	private InventoryIconInfo getInventoryIconInfoFromImage(BufferedImage inventoryIconImage)
+	{
+		int minX = inventoryIconImage.getWidth(), minY = inventoryIconImage.getHeight(), maxX = 0, maxY = 0;
+
+		int pixelCount = 0;
+		double centroidX = 0;
+		double centroidY = 0;
+		Point2D centroid = null;
+		double radius = 0;
+
+		for (int y = 0; y < (inventoryIconImage.getHeight() - 1); y++)
+		{
+			for (int x = 0; x < (inventoryIconImage.getWidth() - 1); x++)
+			{
+				int pixel = inventoryIconImage.getRGB(x, y);
+				int pixelAlpha = pixel >>> 24;
+				if (pixelAlpha == 0)
+				{
+					continue;
+				}
+
+				minX = Math.min(minX, x);
+				minY = Math.min(minY, y);
+				maxX = Math.max(maxX, x);
+				maxY = Math.max(maxY, y);
+
+				if (config.getCooldownIndicatorMode() == CooldownIndicatorMode.PIE)
+				{
+					centroidX += x;
+					centroidY += y;
+					pixelCount++;
+					centroid = new Point2D.Double(centroidX / pixelCount, centroidY / pixelCount);
+					radius = Math.max(radius, centroid.distanceSq(x, y));
```

Comment:
I believe this is comparing radii against a moving target. The centroid is moving from top left to the actual centroid as we are comparing distances to it. This would mean that the further top-left the actual centroid is, the more inaccurate the computed radius is.

Reply:
Oops didn't realise this side effect when consolidating code. Will have a look this week to fix that. Thanks